src/Sylius/Bundle/CoreBundle/Tests/DependencyInjection/SyliusCoreExtensionTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Tests\DependencyInjection;
+
+use Doctrine\Bundle\MigrationsBundle\DependencyInjection\DoctrineMigrationsExtension;
+use Matthias\SymfonyDependencyInjectionTest\PhpUnit\AbstractExtensionTestCase;
+use Sylius\Bundle\CoreBundle\DependencyInjection\SyliusCoreExtension;
+use SyliusLabs\DoctrineMigrationsExtraBundle\DependencyInjection\SyliusLabsDoctrineMigrationsExtraExtension;
+
+final class SyliusCoreExtensionTest extends AbstractExtensionTestCase
+{
+    /**
+     * @test
+     */
+    public function it_autoconfigures_prepending_doctrine_migrations_with_proper_migrations_path_for_test_env(): void
+    {
+        $this->autoconfigurePrependingDoctrineMigrations('test');
+    }
+
+    /**
+     * @test
+     */
+    public function it_autoconfigures_prepending_doctrine_migrations_with_proper_migrations_path_for_test_cached_env(): void
+    {
+        $this->autoconfigurePrependingDoctrineMigrations('test');
+    }
+
+    /**
+     * @test
+     */
+    public function it_autoconfigures_prepending_doctrine_migrations_with_proper_migrations_path_for_dev_env(): void
+    {
+        $this->autoconfigurePrependingDoctrineMigrations('test');
+    }
+
+    protected function getContainerExtensions(): array
+    {
+        return [new SyliusCoreExtension()];
+    }
+
+    private function autoconfigurePrependingDoctrineMigrations(string $env): void
+    {
+        $this->container->setParameter('kernel.environment', $env);
+        $this->container->setParameter('kernel.debug', true);
+
+        $this->container->registerExtension(new DoctrineMigrationsExtension());
+        $this->container->registerExtension(new SyliusLabsDoctrineMigrationsExtraExtension());
+
+        $this->load();
+
+        $extensionConfig = $this->container->getExtensionConfig('doctrine_migrations');
+
+        $this->assertTrue(isset($extensionConfig[0]['migrations_paths']['Sylius\Bundle\CoreBundle\Migrations']));
```

Comment:
You should check also the cases when the `sylius_core.prepend_doctrine_migrations` parameter is false and true, because now I suppose that you are checking the case when it is not set.